### PR TITLE
[WIP] Homepage

### PR DESCRIPTION
**Pull request by filrak**
- Basic setup for HomePage and Capi theme
- Preparation for new theming mechanism (whole mechanism in separate PR to not mix contexts)

### CODE REVIEW

File: src/themes/capybara/pages/Home.vue [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    />
+    <SfSection
+      :heading="{
+        title: 'Best Sellers'
+      }"
+    >
+      <SfCarousel :settings="{ gap: 0 }" class="product-carousel">
+        <SfCarouselItem v-for="(product, i) in newProducts" :key="i">
+          <SfProductCard
+            :title="product.name"
```
Comment (patzick) [inline src/themes/capybara/pages/Home.vue]: why just not take product as prop?

File: src/themes/capybara/pages/Home.vue [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  beforeRouteEnter (to, from, next) {
+    if (!isServer && !from.name) { // Loading products to cache on SSR render
+      next(vm => {
+        vm.$store.dispatch('homepage/fetchNewProducts')
+      })
+    } else {
+      next()
+    }
+  },
+  components: {
```
Comment (patzick) [inline src/themes/capybara/pages/Home.vue]: we should have components above, rocommended way:

- name
- assets (mixins, components, directives, filters)
- props
- data
- computed
- methods
- lifecycle hooks
- render (if used)

source: [The Pope](https://github.com/vuejs/eslint-plugin-vue/issues/13#issuecomment-306128173)

File: src/themes/capybara/store/homepage.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { prepareQuery } from '@vue-storefront/core/modules/catalog/queries/common'
+
+export const homepageStore = {
+  namespaced: true,
+  state: {
+    newProducts: []
+  },
+  mutations: {
+    setNewProducts (state, payload) {
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
    SET_NEW_PRODUCTS (state, payload) {
```

File: src/themes/capybara/store/homepage.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    }
+  },
+  actions: {
+    async fetchNewProducts ({ dispatch, commit }) {
+      const result = await dispatch('product/list', {
+        query: prepareQuery({ queryConfig: 'newProducts' }),
+        size: 8,
+        sort: 'created_at:desc'
+      }, { root: true })
+      commit('setNewProducts', result.items)
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
      commit('SET_NEW_PRODUCTS', result.items)
```

File: src/themes/capybara/store/homepage.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      return result.items
+    }
+  },
+  getters: {
+    getNewProducts: state => {
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
    getNewProducts: state => state.newProducts
```

File: src/themes/capybara/store/homepage.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      return state.newProducts
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
```

File: src/themes/capybara/store/homepage.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    }
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
```

File: src/themes/capybara/store/homepage.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  state: {
+    newProducts: []
+  },
+  mutations: {
+    setNewProducts (state, payload) {
+      state.newProducts = payload
+    }
+  },
+  actions: {
+    async fetchNewProducts ({ dispatch, commit }) {
```
Comment (patzick) [inline src/themes/capybara/store/homepage.ts]:
```suggestion
    async loadNewProducts ({ dispatch, commit }) {
```

File: src/themes/capybara/pages/Home.vue [hunk 9]
```diff
@@ -0,0 +1,273 @@
...
+      heroes: heroData
+    };
+  },
+  computed: {
+    newProducts () {
+      return this.$store.getters['homepage/getNewProducts']
+    }
+  },
+  async asyncData ({ store, route }) {
+    await store.dispatch('homepage/fetchNewProducts')
```
Comment (patzick) [inline src/themes/capybara/pages/Home.vue]:
```suggestion
    await store.dispatch('homepage/loadNewProducts')
```

File: src/themes/capybara/pages/Home.vue [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+  },
+  beforeRouteEnter (to, from, next) {
+    if (!isServer && !from.name) { // Loading products to cache on SSR render
+      next(vm => {
+        vm.$store.dispatch('homepage/fetchNewProducts')
```
Comment (patzick) [inline src/themes/capybara/pages/Home.vue]:
```suggestion
        vm.$store.dispatch('homepage/loadNewProducts')
```

**Comment by filrak** [comment]
It's WIP but some of the comments are useful even in this form, thanks

File: src/themes/capybara/pages/Home.vue [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+              katherina_trn
+            </SfImage>
+          </div>
+          <div class="grid__col">
+            <SfImage src="/assets/homepage/imageB.png">
+              katherina_trn
+            </SfImage>
+          </div>
+        </div>
+        <div class="grid__row">
```
Comment (pkarw) [inline src/themes/capybara/pages/Home.vue]: is there any place where the classes have been pre-defined? @filrak to make it easier for developers to pick the right one

File: src/themes/capybara/pages/Home.vue [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+    <SfCallToAction
+      title="Subscribe to Newsletters"
+      button-text="Subscribe"
+      description="Be aware of upcoming sales and events. Receive gifts and special offers!"
+      class="call-to-action-newsletter"
+      image="/assets/homepage/newsletter.png"
+    />
+    <SfSection
+      :heading="{
+        title: 'Best Sellers'
```
[the remark this answers is not in the document]
Comment (filrak) [reply src/themes/capybara/pages/Home.vue]: what do you mean by worried?

File: src/themes/capybara/pages/Home.vue [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+      }"
+    >
+      <SfCarousel :settings="{ gap: 0 }" class="product-carousel">
+        <SfCarouselItem v-for="(product, i) in newProducts" :key="i">
```
[the remark this answers is not in the document]
Comment (filrak) [reply src/themes/capybara/pages/Home.vue]: imho it will work the same but I can put sku

File: src/themes/capybara/pages/Home.vue (hunk 11 above)
Comment (filrak) [reply src/themes/capybara/pages/Home.vue]: what you mean by predefined?

Comment (filrak) [reply src/themes/capybara/pages/Home.vue]: All classes are encapsulated in Home.vue

**Pull request merged**